### DISCUSSION TITLE
Keep Pull Requests simple!

### DISCUSSION THREAD

**Discussion by stratofax** [cadentcom/pico-wp-block-theme#28] [General]
Small Pull Requests are easier to review and merge. Large PRs that cover many commits and issues are more difficult to review, approve and merge.

Here are some tips to improve the velocity of our workflow on this repo:

- All changes or bug fixes should be tracked with a new Issue.
- Once you start working on an Issue, create a new branch for it, and **link the Issue to the new branch**
- **Don't put multiple issues on the same branch!** Unless several issues involve related changes to the same file or small set of files.
- If your new branch is tracking changes on a small number of files (maybe just one file, even) other developers can work on other files on other branches without creating merge conflicts.
- Because the changes on your new branch will be limited, it will be easier to review, approve, and merge the branch.
- When you start working on an Issue, change the status to _In Progress_ for better project tracking.

**Comment by ratulkhan74** [comment]
@stratofax Okay, I will flow the instruction

**Reply by stratofax** [reply]
Thanks, it will help us close PRs and merge with greater speed.

**Reply by stratofax** [reply]
In fact, let's just apply this simple rule in the future: **each individual issue should be linked to one unique branch and one PR**. We'll try this out as we approach the first release candidate and see how it goes.

If you think two or more Issues should be combined on a single branch and PR, please discuss it in the Issue(s) first before proceeding.

**Reply by ratulkhan74** [reply]
alright

**Comment by stratofax** [comment]
Here are some tips to create good PRs:

A good PR is:

- linked to a single issue
- it's on a separate branch
- all changes are related to the relevant issue

If several issues are related to the same code or set of files, it may make sense to combine them into a single PR. When in doubt, **keep it simple.** Simple PRs are easier to merge and close.

## How to set up a good PR

- Make the name of the PR as descriptive as possible, within a reasonable length. Y
- Link the PR to the relevant issue using the _Development_ section of the PR sidebar (right side of the page). Note that you can also link an Issue to a PR or a new branch on an Issue page, also using the _Development_ section in the sidebar.
- Assign the PR to at least one developer (again, right hand sidebar, in the _Assignees_ section).
- Assign a  _Reviewer_ (right hand sidebar),

These are the essential elements to a good Pull Request

## Additional Info to Improve your PRs

- _Labels:_ as the numbers of Issues and PRs increase, these labels become more useful. Note that you can assign multiple labels to a PR.
- _Projects:_ as this project becomes more complex, take advantage of GitHub's built-in project tracking tools. In many cases, GitHub will automatically update the Status of a related Issue when the linked PR is closed.

Please use these sections in the right-hand sidebar, for both Issues and PRs, to help us track and organize our work.

**Reply by stratofax** [reply]
@ratulkhan74 I put these tips here in the discussion so you can refer to them in the future, and suggest your own ideas.

**Reply by ratulkhan74** [reply]
okay
